### ISSUE TITLE
[QUESTION] Widget to execute arbitrary exes

### ISSUE THREAD

**Issue by 3dfoster** [microsoft/XboxGameBarSamples#72]
Is it possible to create a widget that can run an .exe file if one is provided? I'm wondering if it is possible to develop a shortcut menu so users can pin and launch their games from the Xbox Game Bar.

**Comment by dynamiquel** [comment]
This looks like something you want: https://stackoverflow.com/questions/45911755/how-to-launch-exe-file-in-uwp-app-using-fulltrustlauncher

I don't know too much about UWP but I do believe that the .exe has to be file picked by the user or within your application folder.
An alternative that I know works (at least with Steam) is to use '[Windows.System.Launcher.LaunchUriAsync(Uri)](https://docs.microsoft.com/en-us/uwp/api/windows.system.launcher.launchuriasync?view=winrt-19041)' to launch the game.

Example: `await Windows.System.Launcher.LaunchUriAsync(new Uri("steam://launch/976730"))` would launch Halo: MCC on Steam.

**Comment by vidager** [comment]
Thank you dynamiquel, your answer is correct.  Game Bar currently doesn't provide any new ways to launch other processes, so you'll be limited to all the techniques available to UWP apps in general. 

If the mechanism available to UWP aren't meeting your needs, please let us know via a feature request and we can look at adding something with appropriate protections etc.  We need to honor the Windows privacy/security model, so launching processes would likely require a minimum of one-time user consent or similar.

**Comment by marck7jr** [comment]
@3dfoster there's a way. I'm developing a Game Library with support for Xbox Game Bar and the way I founded to it work with any games without activation protocol support like Steam games were packaging my app with a Win32 application (like WPF Core) using Full Trust restricted capability and using a AppService protocol to make a "link" between the widget and the Win32 application sending which executable and how to launch. If you have in mind to try it, I recommend this series of [articles](https://stefanwick.com/2018/04/06/uwp-with-desktop-extension-part-1/), they helped me a lot when I started to learn how to do it.
